codebundles/sli-incident-threshold/sli.robot

```diff
@@ -0,0 +1,82 @@
+...               When this SLI detects an incident it submits a 1 to denote a signal was sent before returning to 0 when the monitored SLI is healthy.
+Force Tags        SLI    Incident    Threshold
+Suite Setup       Suite Initialization
+Library           RW.Core
+Library           RW.Rest
+Library           RW.Utils
+Library           RW.RunWhen.Papi
+
+*** Keywords ***
+Suite Initialization
+    ${WORKSPACE_NAME}=    RW.Core.Import User Variable    WORKSPACE_NAME
+    ...    type=string
+    ...    description=The workspace the SLI resides in.
+    ...    pattern=\w*
+    ...    example=my-awesome-ws
+    ...    default=my-awesome-ws
+    ${SLX_NAME}=    RW.Core.Import User Variable    SLX_NAME
+    ...    type=string
+    ...    description=The SLX the SLI is attached to.
+    ...    pattern=\w*
+    ...    example=my-awesome-slx
+    ...    default=my-awesome-slx
+    ${HISTORY_WINDOW}=    RW.Core.Import User Variable    HISTORY_WINDOW
+    ...    type=string
+    ...    description=The history window to fetch from the metric store.
+    ...    pattern=((\d+?)d)?((\d+?)h)?((\d+?)m)?((\d+?)s)?
+    ...    example=1h
+    ...    default=1h
+    ${RESOLUTION}=    RW.Core.Import User Variable    RESOLUTION
+    ...    type=string
+    ...    description=The resolution that determines the number of data points.
+    ...    pattern=((\d+?)d)?((\d+?)h)?((\d+?)m)?((\d+?)s)?
+    ...    example=5m
+    ...    default=5m
+    ${SUCCESS_VALUE}=    RW.Core.Import User Variable    SUCCESS_VALUE
+    ...    type=string
+    ...    description=The value searched for in the SLI metrics that indicates a successful / healthy state.
+    ...    pattern=(\d+?)
+    ...    example=1
+    ...    default=1
+    ${EXPECTED_SUCCESS_RATE}=    RW.Core.Import User Variable    EXPECTED_SUCCESS_RATE
+    ...    type=string
+    ...    description=The rate of successes within the data set inspected. If the actual rate is lower than this a TaskSet will be triggered.
+    ...    pattern=(\d+?)
+    ...    example=A rate 0.25 indicates that within a time range of X and success value of Y, we see Y in X's data set 25% of the time.
+    ...    default=0.25
+    ${INCIDENT_TASKSET}=    RW.Core.Import User Variable    INCIDENT_TASKSET
+    ...    type=string
+    ...    description=The name of the SLX containing the TaskSet to run when the monitored SLI is classified as an incident.
+    ...    pattern=\w*
+    ...    example=my-awesome-slx
+    Set Suite Variable    ${WORKSPACE_NAME}    ${WORKSPACE_NAME}
+    Set Suite Variable    ${SLX_NAME}    ${SLX_NAME}
+    Set Suite Variable    ${HISTORY_WINDOW}    ${HISTORY_WINDOW}
+    Set Suite Variable    ${RESOLUTION}    ${RESOLUTION}
+    Set Suite Variable    ${SUCCESS_VALUE}    ${SUCCESS_VALUE}
+    Set Suite Variable    ${EXPECTED_SUCCESS_RATE}    ${EXPECTED_SUCCESS_RATE}
+    Set Suite Variable    ${INCIDENT_TASKSET}    ${INCIDENT_TASKSET}
+
+*** Tasks ***
+Check If SLI Within Incident Threshold
```

Comment:
Overall, we need a way to have a backoff as well... as this will continue to fire the taskset every time the SLI runs; 
![image](https://user-images.githubusercontent.com/18314168/225332717-2679d579-f469-4386-829b-d800cf69fa7a.png)

Reply:
![image](https://user-images.githubusercontent.com/18314168/225332986-df670ecc-6624-4b4b-8a60-3e9a9161b712.png)

Reply:
That's not intended behaviour. I'll bring this up to the team as there should be a builtin cooldown of 8 hours between runs.